.github/workflows/test.yml

```diff
@@ -0,0 +1,59 @@
+name: test
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```

Comment:
What does this do, and why?

(The push one is more obvious, although I'd probably want all my branches tested)

Reply:
These define the triggers, so these jobs will run if there's either a push to master, or a PR targeting master. I can definitely change it to run on push to all branches.

Reply:
I think for this repository I would use all branches - unless perhaps people with forks felt strongly that a different choice was better?

Reply:
No that would be fine. In the past I've limited runs to the `master` branch just because typically development branches will eventually end up in a PR to get tested, but having the dev branches themselves tested may help catch issues earlier. I will push up the changes shortly.